### PR TITLE
feat(core-button-link):  adding a prop to make the button full width

### PR DESCRIPTION
**Pull request by sudhirraivns0103**
I have added prop as on option to make the button full width.
Closes #1461

### CODE REVIEW

File: packages/ButtonLink/ButtonLink.jsx [hunk 1]
```diff
@@ -43,6 +43,20 @@ const StyledButtonLink = styled(StyledButton)(
         color: hoverColor,
       },
     }
+  },
+  ({ fullWidth }) => {
+    let width
+    if (fullWidth === 'true') {
```
Comment (mike-bunce) [inline packages/ButtonLink/ButtonLink.jsx:49]: Can you write this as a ternary statement instead?

File: e2e/visual/components/CartesianButtonLink.jsx [hunk 2]
```diff
@@ -12,6 +12,7 @@ const CartesianButtonLink = props => (
     variant={['primary', 'secondary', 'inverted']}
     href="http://tds.telus.com"
     children="TELUS Design System"
+    fullWidth="false"
```
Comment (mike-bunce) [inline e2e/visual/components/CartesianButtonLink.jsx:15]: Can you write this as requested: fullWidth={[false,true]}

File: packages/ButtonLink/ButtonLink.jsx (hunk 1 above)
Comment (sudhirraivns0103) [reply packages/ButtonLink/ButtonLink.jsx]: Hi @mike-bunce,
I have updated PR again with requested changes

File: packages/ButtonLink/ButtonLink.jsx [hunk 3]
```diff
@@ -87,6 +98,11 @@ ButtonLink.propTypes = {
    * Target URL
    */
   href: PropTypes.string,
+  /**
+   * If `true`, sets `ButtonLink` to 100% width.
+   * @since 2.2.0
+   */
+  fullWidth: PropTypes.oneOf(['true', 'false']),
```
Comment (jraff) [inline packages/ButtonLink/ButtonLink.jsx:105]: Can we use a boolean instead of true/false strings?

Comment (sudhirraivns0103) [reply packages/ButtonLink/ButtonLink.jsx]: Hi @jraff,
I have changed boolean instead of true/false

File: packages/ButtonLink/ButtonLink.jsx [hunk 4]
```diff
@@ -43,6 +43,14 @@ const StyledButtonLink = styled(StyledButton)(
         color: hoverColor,
       },
     }
+  },
+  ({ fullWidth }) => {
+    const width = fullWidth === 'true' ? '100%' : 'auto'
```
Comment (jraff) [inline packages/ButtonLink/ButtonLink.jsx:48]: This line should be changed to `const width = fullWidth ? '100%' : 'auto'` as `fullWidth` should no longer be a string.

Comment (sudhirraivns0103) [reply packages/ButtonLink/ButtonLink.jsx]: Hi @jraff,
I have update your requested change.

**Pull request merged**